src/beta_inc.jl

```diff
@@ -1001,7 +1001,7 @@ function _beta_inc_inv(a::Float64, b::Float64, p::Float64, q::Float64=1-p)
     while true
         p_approx = beta_inc(a, b, x)[1]
         xin = x
-        p_approx = (p_approx - p)*min(floatmax(), exp(lb + r*log(xin) + t*log1p(-xin)))
+        p_approx = (p_approx - p)*min(floatmax(), exp(lb + LogExpFunctions.xlogy(r, xin) + t*log1p(-xin)))
```

Comment:
I wonder if we should apply the same fix to the last term?
```suggestion
        p_approx = (p_approx - p)*min(floatmax(), exp(lb + LogExpFunctions.xlogy(r, xin) + LogExpFunctions.xlog1py(t, -xin)))
```
We might have to adjust the LogExpFunctions compat entry for it.

Reply:
Indeed. Only with #392, I was able to trigger a similar issue in the opposite direction. I've added a test and adjusted the compat bound.